x-pack/plugins/security_solution/public/timelines/components/timeline/esql_tab_content/index.tsx

```diff
@@ -238,27 +236,16 @@ export const DiscoverTabContent: FC<DiscoverTabContentProps> = ({ timelineId })
         savedSearchAppState = getAppStateFromSavedSearch(localSavedSearch);
       }
 
-      const defaultDiscoverAppState = await getDefaultDiscoverAppState();
-
       const finalAppState =
         savedSearchAppState?.appState ?? discoverAppState ?? defaultDiscoverAppState;
 
-      const urlAppState = stateContainer.appState.isEmptyURL()
-        ? undefined
-        : stateContainer.appState.getState();
-
-      const hasESQLURlState = urlAppState?.query && 'esql' in urlAppState.query;
-
-      /*
-       * Url state should NOT apply if there is already a saved search being loaded
-       * */
-      const shouldApplyESQLUrlState = !savedSearchAppState?.appState && hasESQLURlState;
-
-      if (!shouldApplyESQLUrlState) {
-        /*
-         * If url state applies, it should be a no-op and there is no need to update the state container.
-         * Discover should automatically pick up url state
-         * */
+      if (stateContainer.appState.isEmptyURL()) {
+        if (savedSearchAppState?.savedSearch.timeRange) {
+          stateContainer.globalState.set({
+            ...stateContainer.globalState.get(),
+            time: savedSearchAppState?.savedSearch.timeRange,
+          });
```

Comment:
This is the edge case issue but because of removing below line, if URL has `non-esql` state, it will also be restored as show below.  Below line was written so that we can we can ignore any other state if not `esql`

`const hasESQLURlState = urlAppState?.query && 'esql' in urlAppState.query;`

I think if url has non-esql state, we should restore `finalAppState`

![Screenshot 2024-03-19 at 09 11 35](https://github.com/elastic/kibana/assets/7485038/36c31b65-81a4-4b06-b19e-e1a0cdad4e96)

Reply:
@logeekal reverted code related to that, could you please verify? 🙏